pkg/api/validation/validation.go

```diff
@@ -1765,6 +1765,23 @@ func validateEndpointAddress(address *api.EndpointAddress) errs.ValidationErrorL
 	return allErrs
 }
 
+func ValidateIpIsNotLinkLocalOrLoopback(ipAddress string) errs.ValidationErrorList {
```

Comment:
This should be called from ValidateService() in this file, which currently tests DeprecatedPublicIPs for 0.0.0.0 and loopback